test/IDE/complete_escaped_keyword.swift

```diff
@@ -27,7 +27,7 @@ enum MyEnum {
 // STATIC_PRIMARY: Begin completion
 // STATIC_PRIMARY-DAG: Decl[LocalVar]/Local:               self[#MyEnum.Type#]; name=self
 // STATIC_PRIMARY-DAG: Decl[EnumElement]/CurrNominal:      `class`({#struct: String#})[#MyEnum#]; name=`class`(struct: String)
-// STATIC_PRIMARY-DAG: Decl[EnumElement]/CurrNominal:      `let`({#`var`: String#})[#MyEnum#]; name=`let`(`var`: String)
+// STATIC_PRIMARY-DAG: Decl[EnumElement]/CurrNominal:      `let`({#var: String#})[#MyEnum#]; name=`let`(var: String)
```

Comment:
Do you know where/how these results are generated? It would be great to keep backticks in place, otherwise that would generate a warning when picked.

Reply:
I thought that this was generated completions at the call site and we don't produce warnings when var is used unescaped at the call site. So for example:

```swift
func foo(var: Int) {} // only this generates a warning
func bar(`var`: Int) {}

foo(var: 1)
bar(var: 1)
```

I might be wrong about these being call site completions though.

Reply:
Even if that's the case I think we should always produce a warning which suggests to escape the use of `var` or `let` regardless if it's in the declaration or a call.

Reply:
Actually let me step back a bit, I think we should have it consistent with other keywords in the call site, if e.g. `class` or `struct` don't suggest backticks `let` & `var` shouldn't too for consistency.

Reply:
Final decision on this: We decided to not produce the warning at the call site.

It’s useful to give the warning in declarations because there’s the possible ambiguity regarding the behavior, but at the call site that ambiguity doesn’t exist. Providing that warning also makes it hard for people to write nice apis if they have a good reason to use var / let as a keyword. Additionally, that's consistent with the way we treat other similar keywords. For example the following produces no warnings:

```swift
func foo(`class`: Int) { print(`class`)}

foo(class: 3)
```

Reply:
+1